pkg/util/config/feature_config.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+)
+
+const (
+	// known keys
+	enableAllAlphaFeatures = "enableAllAlphaFeatures"
+)
+
+var featureDefaults = map[string]bool{
+	enableAllAlphaFeatures: false,
+}
+
+type FeatureConfig interface {
```

Comment:
Yes.  Gate is better in that context, too :)

On Wed, Aug 17, 2016 at 9:06 PM, Jeff Lowdermilk <notifications@github.com>
wrote:

> In pkg/util/config/feature_config.go
> <https://github.com/kubernetes/kubernetes/pull/30468#discussion_r75246152>
> :
>
> > +	"strings"
> > +
> > +	"github.com/golang/glog"
> > +	"github.com/spf13/pflag"
> > +)
> > +
> > +const (
> > +	// known keys
> > +	enableAllAlphaFeatures = "enableAllAlphaFeatures"
> > +)
> > +
> > +var featureDefaults = map[string]bool{
> > +	enableAllAlphaFeatures: false,
> > +}
> > +
> > +type FeatureConfig interface {
>
> If others agree it's strictly better I'll change it, but imo the flag name
> should match (--feature-gate? --feature-gates?)
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/30468/files/26946965fe3c4383f56daef9c62aa74a7542035b#r75246152>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFVgVMHLreJcL4NrJHEQdKMBTUHPVlEDks5qg9o_gaJpZM4JieFK>
> .
>